### ISSUE TITLE
finishconnect() failed: Connection refused.

### ISSUE THREAD

**Issue by therealOri** [Snowiiii/Pumpkin#138]
- [x] I've searched existing issues and couldn't find a duplicate.

**Describe the bug**
Unable to connect to server after it comes online. finishconnect() failed.
__ __

<br>

**To Reproduce**
- git clone https://github.com/Snowiiii/Pumpkin.git
- cd Pumpkin
- cargo run --release
__ __

<br>

**Expected behavior**
Join server hosted on 0.0.0.0:25565

**Environment (please complete the following information):**
- Operating System: [CachyOS/Arch]
- Server Software Version/Commit: [version = "0.1.0" - commit: 5ed7cae]
__ __

<br>

**Additional context**
Everything is default and not modified in terms of settings/configs. No vanilla world anything either.
Rust version: rustc 1.81.0 (eeb90cda1 2024-09-04)

Logs:
```bash
2024-10-16T21:43:02.009Z INFO  [pumpkin::server@main] Loading Plugins
2024-10-16T21:43:02.009Z WARN  [pumpkin_world::level@main] Pumpkin currently only supports Superflat World generation. Use a vanilla ./world folder to play in a normal world.
2024-10-16T21:43:02.020Z INFO  [pumpkin@main] Started Server took 11ms
2024-10-16T21:43:02.020Z INFO  [pumpkin@main] You now can connect to the server, Listening on 0.0.0.0:25565
2024-10-16T21:43:03.285Z INFO  [pumpkin@main] Accepted connection from: xxx.x.x.x:xxxxx
[pumpkin/src/client/client_packet.rs:35:9] "handshake" = "handshake"
[pumpkin/src/client/client_packet.rs:61:9] "ping" = "ping"
```
__ __

**Comment by Snowiiii** [comment]
Hey, With what ip your trying to connect using your client ?

**Comment by therealOri** [comment]
Accepted connection from: 127.0.0.1:50000 (No idea how or why this is chosen but it is.)

However, interesting note. I don't even try to click join server yet. It does all that the moment I click onto "Multiplayer"

**Comment by Snowiiii** [comment]
> Accepted connection from: 127.0.0.1:50000 (No idea how or why this is chosen but it is.)
> 
> However, interesting note. I don't even try to click join server yet. It does all that the moment I click onto "Multiplayer"

Hey, I was asking what ip your using to join your server, I heard multiple reports of joining with 127.0.0.1 or localhost does work while 0.0.0.0 does not, maybe try that out

When you Open the Multiplayer screen your Client sends a Packet to every Server in the Server list, He wants to see the Status (MOTD) and Ping

**Comment by therealOri** [comment]
Tried setting the server up to use 127.0.0.1 instead of 0.0.0.0 and it's the same error.
![image](https://github.com/user-attachments/assets/10d0985b-02c2-4d3c-bb26-c75d5d870dab)





Logs:
```bash
2024-10-16T22:02:58.889Z INFO  [pumpkin::server@main] Loading Plugins
2024-10-16T22:02:58.889Z WARN  [pumpkin_world::level@main] Pumpkin currently only supports Superflat World generation. Use a vanilla ./world folder to play in a normal world.
2024-10-16T22:02:58.905Z INFO  [pumpkin@main] Started Server took 16ms
2024-10-16T22:02:58.905Z INFO  [pumpkin@main] You now can connect to the server, Listening on 127.0.0.1:25565
2024-10-16T22:03:26.648Z INFO  [pumpkin@main] Accepted connection from: 127.0.0.1:45822 (this is different now?)
[pumpkin/src/client/client_packet.rs:35:9] "handshake" = "handshake"
[pumpkin/src/client/client_packet.rs:61:9] "ping" = "ping"
```

**Comment by Snowiiii** [comment]
Im not talking about the server, im talking about the client
![image](https://github.com/user-attachments/assets/cd1f4531-0951-494f-9416-0a3a8bdb3918)

**Comment by therealOri** [comment]
Ah, the direct connection server address was for a different server.

I thought I could just add a server, use the local host ip:port and join that way. Didn't know You could only join via direct connection.

When I join the port/server using 127.0.0.1 it worked and I joined.

**Comment by Snowiiii** [comment]
Great, I wonder why some people are having this issue, I and most devs do not have this issue

**Comment by Snowiiii** [comment]
any idea?, is 0.0.0.0 maybe a bad default port

**Comment by therealOri** [comment]
I do not know, may be a rust thing. I don't know a lot about rust yet so I couldn't say.

I do also timeout after sitting afk for like 1min. But That can be a different issue all on its own.

**Comment by Snowiiii** [comment]
> I do not know, may be a rust thing. I don't know a lot about rust yet so I couldn't say.
> 
> I do also timeout after sitting afk for like 1min. But that can be a different issue all on its own.

Yep, a known issue :/. Has to do with the blocking issue we have

**Comment by therealOri** [comment]
Okay, then I believe this issue can be closed as it is an issue using 0.0.0.0 instead of 127.0.0.1.
